core/src/main/scala/io/finch/EndpointResult.scala

```diff
@@ -45,6 +45,11 @@ sealed abstract class EndpointResult[F[_], +A] {
     case _ => None
   }
 
+  final def map[B](fn: A => B)(implicit F: Effect[F]): EndpointResult[F, B] = this match {
```

Comment:
Why this is needed?

Reply:
I tried to make `streamJsonBody` simpler. We still need the access to `Input.request.charsetOrUtf8` to run the decoder, so it's impossible to just use `streamBody.map(a => b)`.
Check implementation of this endpoint.